src/field_impl.h

```diff
@@ -336,4 +336,32 @@ SECP256K1_INLINE static void secp256k1_fe_from_storage(secp256k1_fe *r, const se
 #endif
 }
 
+static void secp256k1_fe_impl_inv(secp256k1_fe *r, const secp256k1_fe *x);
+SECP256K1_INLINE static void secp256k1_fe_inv(secp256k1_fe *r, const secp256k1_fe *x) {
+#ifdef VERIFY
+    int input_is_zero = secp256k1_fe_normalizes_to_zero(x);
+#endif
+    secp256k1_fe_impl_inv(r, x);
+#ifdef VERIFY
+    VERIFY_CHECK(secp256k1_fe_normalizes_to_zero(r) == input_is_zero);
+    r->magnitude = x->magnitude > 0;
+    r->normalized = 1;
+    secp256k1_fe_verify(r);
+#endif
+}
+
+static void secp256k1_fe_impl_inv_var(secp256k1_fe *r, const secp256k1_fe *x);
+SECP256K1_INLINE static void secp256k1_fe_inv_var(secp256k1_fe *r, const secp256k1_fe *x) {
+#ifdef VERIFY
+    int input_is_zero = secp256k1_fe_normalizes_to_zero(x);
+#endif
+    secp256k1_fe_impl_inv_var(r, x);
+#ifdef VERIFY
+    VERIFY_CHECK(secp256k1_fe_normalizes_to_zero(r) == input_is_zero);
+    r->magnitude = x->magnitude > 0;
```

Comment:
Before this PR, how could `r->magnitude` have ended up as 0?

Reply:
It couldn't; the new logic is slightly more precise.

Reply:
Makes sense now, I didn't realize before that `secp256k1_fe_impl_inv{_var}` returns zero if the input was zero.